public/js/reducers/sidebars.js

```diff
@@ -0,0 +1,69 @@
+// @flow
+const I = require("immutable");
+const makeRecord = require("../utils/makeRecord");
+const C = require("../constants");
+
+import type { Action } from "../actions/types";
+import type { Record } from "../utils/makeRecord";
+
+type SidebarState = {
+  collapsed: boolean,
+  width: number,
+  prevWidth: number,
+};
+
+type SidebarsState = {
+  left: I.Map<string, SidebarState>,
+  right: I.Map<string, SidebarState>,
+};
+
+const State = makeRecord(({
+  left: I.Map({
+    collapsed: false,
+    width: 300,
+    prevWidth: 0
```

Comment:
I think it should be possible to keep `collapsed` and `width` and be okay:

e.g.
* app starts width is 300
* user resizes width is 370
* user collapses width is still 370, collapsed true
* user expands  width is still 370, collapsed false
* ...